src/resources/events/eventsApi.ts

```diff
@@ -56,11 +58,28 @@ export class EventsAPI extends BaseResourceAPI<CogniteEvent> {
    * ```
    */
   public aggregate = (
-    query: EventAggregateQuery
-  ): Promise<EventAggregate[]> => {
+    query: EventCountAggregate
+  ): Promise<CountAggregateResponse[]> => {
     return super.aggregateEndpoint(query);
   };
 
+  /**
+   * [Aggregate events](https://docs.cognite.com/api/v1/#operation/aggregateEvents)
+   *
+   * ```js
+   * const aggregates = await client.events.uniqueValuesAggregate({ filter: { assetIds: [1, 2, 3] }, fields: ['subtype'] });
+   * console.log('Unique values: ', aggregates)
+   * ```
+   */
+  public uniqueValuesAggregate = (
+    query: EventUniqueValuesAggregate
```

Comment:
this is a mess, we cannot have it.
it's not java, we don't need to complicate things

Reply:
also this kills any idea of having even semi generated sdks

Reply:
Why? Without it then the return type is really messy since it is dependent on the function call.
We discussed also the semi generated sdks scenario with this approach with @hhenriques